test/peripheral/551_exportProxy.ts

```diff
@@ -0,0 +1,159 @@
+  let proxyRegistry: Contract
+  let dsProxy: Contract
+
+  beforeEach(async () => {
+    const block = await web3.eth.getBlockNumber()
+    maturity1 = (await web3.eth.getBlock(block)).timestamp + 30000000 // Far enough so that the extra weth to borrow is above dust
+
+    env = await YieldEnvironmentLite.setup([maturity1])
+    controller = env.controller
+    vat = env.maker.vat
+    dai = env.maker.dai
+    weth = env.maker.weth
+
+    fyDai1 = env.fyDais[0]
+
+    // Setup Pool
+    pool1 = await Pool.new(dai.address, fyDai1.address, 'Name', 'Symbol', { from: owner })
+
+    // Setup ExportProxy
+    exportProxy = await ExportProxy.new(controller.address, [pool1.address], { from: owner })
+
+    // Allow owner to mint fyDai the sneaky way, without recording a debt in controller
+    await fyDai1.orchestrate(owner, id('mint(address,uint256)'), { from: owner })
+
+    // Initialize Pool1
+    const daiReserves = bnify(daiTokens1).mul(5)
+    await env.maker.getDai(owner, daiReserves, rate1)
+    await dai.approve(pool1.address, daiReserves, { from: owner })
+    await pool1.mint(owner, owner, daiReserves, { from: owner })
+
+    // Add fyDai
+    const additionalFYDaiReserves = bnify(fyDaiTokens1).mul(2)
+    await fyDai1.mint(owner, additionalFYDaiReserves, { from: owner })
+    await fyDai1.approve(pool1.address, additionalFYDaiReserves, { from: owner })
+    await pool1.sellFYDai(owner, owner, additionalFYDaiReserves, { from: owner })
+
+    // Setup DSProxyFactory and DSProxyCache
+    proxyFactory = await DSProxyFactory.new({ from: owner })
+
+    // Setup DSProxyRegistry
+    proxyRegistry = await DSProxyRegistry.new(proxyFactory.address, { from: owner })
+
+    // Sets DSProxy for user
+    await proxyRegistry.build({ from: user })
+    dsProxy = await DSProxy.at(await proxyRegistry.proxies(user))
+  })
+
+  it('does not allow to execute the flash mint callback to users', async () => {
+    const data = web3.eth.abi.encodeParameters(
+      ['address', 'address', 'uint256', 'uint256'],
+      [pool1.address, user, 1, 0]
+    )
+    await expectRevert(exportProxy.executeOnFlashMint(1, data, { from: user }), 'ExportProxy: Restricted callback')
+  })
+
+  it('does not allow to move more debt than existing in env', async () => {
+    await expectRevert(
+      exportProxy.exportPosition(pool1.address, wethTokens1, fyDaiTokens1, { from: user }),
+      'ExportProxy: Not enough debt in Yield'
+    )
+  })
+
+  it('does not allow to move more weth than posted in env', async () => {
+    await env.postWeth(user, wethTokens1)
+    const toBorrow = (await env.unlockedOf(WETH, user)).toString()
+    await controller.borrow(WETH, maturity1, user, user, toBorrow, { from: user })
+
+    await expectRevert(
+      exportProxy.exportPosition(pool1.address, bnify(wethTokens1).mul(2), toBorrow, { from: user }),
+      'ExportProxy: Not enough collateral in Yield'
+    )
+  })
+
+  it('moves yield vault to maker', async () => {
```

Comment:
Can we add a test that shows that a malicious user cannot force-migrate another user's vault back to maker?

Reply:
`exportPosition` doesn't take an `user` argument, it moves the vault of the caller.

There is a test ensuring `executeOnFlashMint` can only be called by a fyDai contract, and `_exportPosition` is internal.